opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -1625,44 +1620,34 @@ public void applyConfig(String configuration, boolean reindex, boolean interacti
      * @param config the incoming configuration
      * @param reindex is the message result of reindex
      * @param interactive true if in interactive mode
-     *
      */
     public void applyConfig(Configuration config, boolean reindex, boolean interactive) {
-        try {
-            configLock.writeLock().lock();
-
-            setConfiguration(config, interactive);
-            LOGGER.log(Level.INFO, "Configuration updated: {0}",
-                    getConfigurationValue("sourceRoot"));
-
-            if (reindex) {
-                // We are assuming that each update of configuration
-                // means reindex. If dedicated thread is introduced
-                // in the future solely for the purpose of getting
-                // the event of reindex, the 2 calls below should
-                // be moved there.
-                refreshSearcherManagerMap();
-                maybeRefreshIndexSearchers();
-                // Force timestamp to update itself upon new config arrival.
-                refreshDateForLastIndexRun();
-            }
+        setConfiguration(config, interactive);
+        LOGGER.log(Level.INFO, "Configuration updated");
+
+        if (reindex) {
+            // We are assuming that each update of configuration means reindex. If dedicated thread is introduced
+            // in the future solely for the purpose of getting the event of reindex, the 2 calls below should
+            // be moved there.
+            refreshSearcherManagerMap();
+            maybeRefreshIndexSearchers();
+            // Force timestamp to update itself upon new config arrival.
+            refreshDateForLastIndexRun();
+        }
 
-            // start/stop the watchdog if necessary
-            if (isAuthorizationWatchdog() && config.getPluginDirectory() != null) {
-                watchDog.start(new File(config.getPluginDirectory()));
-            } else {
-                watchDog.stop();
-            }
+        // start/stop the watchdog if necessary
+        if (isAuthorizationWatchdog() && config.getPluginDirectory() != null) {
```

Comment:
`config.getPluginDirectory()` this is ok without locking?

Reply:
also there are more calls like this here (getPluginStack, messageLimit)

Reply:
answer this and i think we're ready to go

Reply:
the configuration is already set so using locked versions is fine and actually necessary.